sdk/storage/azure-storage-datamovement/src/job_engine_task_generation.cpp

```diff
@@ -127,7 +128,63 @@ namespace Azure { namespace Storage { namespace _detail {
     }
     else if (m_model.Source.m_type == _internal::TransferEnd::EndType::AzureBlobFolder)
     {
-      AZURE_NOT_IMPLEMENTED();
+      const std::string rootDirectory = _internal::PathFromUrl(m_model.Destination.m_url);
+      std::string currentDirectory = rootDirectory;
+
+      partGens.push_back(gen);
+      if (!_internal::IsDirectory(currentDirectory))
+      {
+        _internal::CreateDirectory(currentDirectory);
+        // TODO: error handling
+      }
+
+      do
+      {
+        PartGeneratorModel currGen = std::move(partGens.back());
+        partGens.pop_back();
+
+        Blobs::ListBlobsOptions options;
+        options.Prefix = m_model.Source.m_blobFolder.Value().GetFolderPath();
```

Comment:
Does return value of `.GetFolderPath()` have `/` at the end already? If not, we need to add a slash here, so as to make sure only blobs in the v-directory are listed.

Reply:
Updated